connectors/sources/confluence.py

```diff
@@ -177,6 +178,25 @@ async def paginated_api_call(self, url_name, **url_kwargs):
                 )
                 break
 
+    async def verify_spaces(self):
+        """Checks if user configured spaces are available in confluence
+
+        Raises:
+            Exception: Configured unavailable spaces: <unavailable_space_keys>
+        """
+        if self.spaces == ["*"]:
+            return
+        space_keys = []
+        async for response in self.paginated_api_call(
+            url_name=SPACE, api_query=SPACE_QUERY
+        ):
+            spaces = response.get("results", [])
+            space_keys.extend([space["key"] for space in spaces])
+        if unavailable_spaces := set(self.spaces) - set(space_keys):
+            raise Exception(
+                f"Configured unavailable spaces: {', '.join(unavailable_spaces)}"
```

Comment:
I would also change the error to something like.

Spaces "{', '.join(unavailable_spaces)}" are not available. Available spaces are: {', '.join(space_keys)}